File: backend/src/backend/Domain/Entities/UserProfile.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Domain.Entities;

[Table("UserProfile")]
public partial class UserProfile : BaseEntity<long>
{

    [ForeignKey(nameof(User))]
    public long UserId { get; set; }

    [StringLength(150)]
    public string? FullName { get; set; }

    [Column(TypeName = "date")]
    public DateTime? Dob { get; set; }

    [StringLength(10)]
    [Unicode(false)]
    public string? Gender { get; set; }

    [StringLength(255)]
    public string? AvatarUrl { get; set; }

    [StringLength(200)]
    public string? AvatarPublicId { get; set; }

    public int? ProvinceCode { get; set; }
    [StringLength(50)]
    public string? ProvinceName { get; set; }
    public int? WardCode { get; set; }

    [StringLength(100)]
    public string? WardName { get; set; }

    public string? AddressDetail { get; set; }

    [StringLength(150)]
    public string? EmergencyName { get; set; }

    [Required]
    [StringLength(15)]
    public string EmergencyPhone { get; set; } // ĐÃ SỬA: string

    [Required]
    [StringLength(15)]
    public string NumberCard { get; set; } // ĐÃ SỬA: string

    [StringLength(150)]
    public string? PlaceOfOrigin { get; set; }

    [StringLength(150)]
    public string? PlaceOfResidence { get; set; }

    [Precision(7)]
    public DateTimeOffset? IssuedDate { get; set; }

    [Precision(7)]
    public DateTimeOffset? ExpiryDate { get; set; }

    [StringLength(100)]
    public string? IssuedBy { get; set; }

    [Required]
    [Precision(0)]
    public DateTimeOffset CreatedAt { get; set; }

    [Required]
    [Precision(0)]
    public DateTimeOffset UpdatedAt { get; set; }

    public virtual AspNetUser User { get; set; } = null!;
}

File: frontend/src/types/userProfile.ts
import type { RentalHistoryDTO } from "./rental.history";

export interface UserProfileDTO {
userId: number;
isVerify?: string | null;
fullName?: string | null;
phoneNumber?: string | null;
email?: string | null;
dob?: string | null;
gender?: string | null;
avatarUrl?: string | null;
emergencyName?: string | null;
emergencyPhone: string;
provinceCode?: number | null;
provinceName?: string | null;
wardCode?: number | null;
wardName?: string | null;
addressDetail?: string | null;
numberCard: string;
placeOfOrigin?: string | null;
placeOfResidence?: string | null;
issuedDate?: string | null;
expiryDate?: string | null;
issuedBy?: string | null;

totalDistanceKm: number;
totalTrips: number;
totalDurationMinutes: number;
totalCo2SavedKg: number;
totalCaloriesBurned: number;

rentalHistories: RentalHistoryDTO[];
createdAt: string;
updatedAt: string;
}

export type UpdateUserProfileBasicDTO = {
fullName?: string | null;
avatarUrl?: string | null;
emergencyName : string;
emergencyPhone: string;
addressDetail?: string | null;
dob?: string | null;
gender?: string | null;
phoneNumber?: string | null;
};